src/linker/Linker.Steps/SweepStep.cs

```diff
@@ -531,6 +531,14 @@ void SweepDebugInfo (Collection<MethodDefinition> methods)
 
 		protected virtual bool ShouldRemove<T> (T element) where T : IMetadataTokenProvider
 		{
+			if (typeof (T) == typeof (ExportedType)) {
+				var typeDef = (element as ExportedType).Resolve ();
+				if (Context.KeepTypeForwarderOnlyAssemblies)
+					return !Annotations.IsMarked (typeDef);
+
+				return !Annotations.IsMarked (typeDef) || !Annotations.IsMarked (element);
```

Comment:
What is the rationale behind keeping forwarders that are never used?

Reply:
Changed it -- we just keep forwarders when both the forwarder and the type it points to are marked (unless we use `--keep-forwarders`, in which case we keep the forwarder iff the type it points to is marked).